sklearn/linear_model/_base.py

```diff
@@ -287,11 +315,33 @@ def predict(self, X):
 
     def _set_intercept(self, X_offset, y_offset, X_scale):
         """Set the intercept_"""
+
+        xp, _ = get_namespace(X_offset, y_offset, X_scale)
+
         if self.fit_intercept:
             # We always want coef_.dtype=X.dtype. For instance, X.dtype can differ from
             # coef_.dtype if warm_start=True.
-            self.coef_ = np.divide(self.coef_, X_scale, dtype=X_scale.dtype)
-            self.intercept_ = y_offset - np.dot(X_offset, self.coef_.T)
+            coef_ = self.coef_ = xp.divide(
+                xp.astype(self.coef_, X_scale.dtype), X_scale
```

Comment:
Setting `copy=False` explicitly is like a runtime check that a copy is actually not made: it will error out if copyless is impossible. The current (which uses the default value, which is `copy=None`) also tries to avoid a copy, but it falls back to a copy if necessary. 

So from your comment:

> We can probably spare a memory copy here:

I think `copy=None` is good already ? and it has the advantage to be less verbose and remain flexible with possibly weird array namespaces.

Reply:
Mistake from me, I forgot that was about `xp.astype` and not `xp.asarray`. You're right, `copy=False` is the appropriate code here (see https://data-apis.org/array-api/latest/API_specification/generated/array_api.astype.html#astype ).